pkg/operator/apis/monitoring/v1/types.go

```diff
@@ -1405,6 +1417,201 @@ type NodeMonitoring struct {
 	Spec NodeMonitoringSpec `json:"spec"`
 }
 
+// NodeScrapeEndpoint specifies a node's Prometheus metrics endpoint to scrape.
+type NodeScrapeEndpoint struct {
+	// HTTP path to scrape metrics from. Defaults to "/metrics".
+	Path string `json:"path,omitempty"`
+	// Interval at which to scrape metrics. Must be a valid Prometheus duration.
+	// +kubebuilder:validation:Pattern="^((([0-9]+)y)?(([0-9]+)w)?(([0-9]+)d)?(([0-9]+)h)?(([0-9]+)m)?(([0-9]+)s)?(([0-9]+)ms)?|0)$"
+	// +kubebuilder:default="1m"
+	Interval string `json:"interval"`
+	// Timeout for metrics scrapes. Must be a valid Prometheus duration.
+	// Must not be larger then the scrape interval.
+	Timeout string `json:"timeout,omitempty"`
+	// Relabeling rules for metrics scraped from this endpoint. Relabeling rules that
+	// override protected target labels (project_id, location, cluster, namespace, job,
+	// instance, or __address__) are not permitted. The labelmap action is not permitted
+	// in general.
+	MetricRelabeling []RelabelingRule `json:"metricRelabeling,omitempty"`
+}
+
+func (n *NodeMonitoring) GetKey() string {
+	return fmt.Sprintf("NodeMonitoring/%s", n.Name)
+}
+
+func (n *NodeMonitoring) GetEndpoints() []NodeScrapeEndpoint {
+	return n.Spec.Endpoints
+}
+
+func (nm *NodeMonitoring) ValidateCreate() error {
+	if len(nm.Spec.Endpoints) == 0 {
+		return errors.New("at least one endpoint is required")
+	}
+	// TODO(freinartz): extract validator into dedicated object (like defaulter). For now using
+	// example values has no adverse effects.
+	_, err := nm.ScrapeConfigs("test_project", "test_location", "test_cluster")
+	return err
+}
+
+func (nm *NodeMonitoring) ValidateUpdate(old runtime.Object) error {
+	// Validity does not depend on state changes.
+	return nm.ValidateCreate()
+}
+
+func (nm *NodeMonitoring) ValidateDelete() error {
+	// Deletions are always valid.
+	return nil
+}
+
+func (nm *NodeMonitoring) ScrapeConfigs(projectID, location, cluster string) (res []*promconfig.ScrapeConfig, err error) {
+	for i := range nm.Spec.Endpoints {
+		c, err := nm.endpointScrapeConfig(i, projectID, location, cluster)
+		if err != nil {
+			return nil, fmt.Errorf("invalid definition for endpoint with index %d: %w", i, err)
+		}
+		res = append(res, c)
+	}
+	return res, nil
+}
+
+func (nm *NodeMonitoring) endpointScrapeConfig(index int, projectID, location, cluster string) (*promconfig.ScrapeConfig, error) {
+	// Filter targets that belong to selected nodes.
+	relabelCfgs, err := relabelingsForSelector(nm.Spec.Selector, nm)
+	if err != nil {
+		return nil, err
+	}
+
+	relabelCfgs = append(relabelCfgs,
```

Comment:
idea: hmmm I wonder if there's no opportunity to reuse some code between Pod/Cluster monitorings. Not a blocker (: